# nios2: do not gate syscall restart on the user r1 register.
# Fixes ERESTARTNOHAND (errno 514) leaking from pselect6 and waitpid
# (pselect/popen tests). Full description + reproducer in
# uclibc-ng/git/kernel-submit/0001-nios2-do-not-gate-syscall-restart-*.patch
--- linux-6.5.10/arch/nios2/kernel/signal.c	2023-11-02 09:37:00.000000000 +0100
+++ ../linux-6.5.10/arch/nios2/kernel/signal.c	2026-06-09 21:53:45.839559034 +0200
@@ -242,7 +242,7 @@
 	/*
 	 * If we were from a system call, check for system call restarting...
 	 */
-	if (regs->orig_r2 >= 0 && regs->r1) {
+	if (regs->orig_r2 >= 0) {
 		continue_addr = regs->ea;
 		restart_addr = continue_addr - 4;
 		retval = regs->r2;
